### ISSUE TITLE
Provide more explicit warnings when using Material 2 widgets with Material 3 Theme

### ISSUE THREAD

**Issue by khanhnwin** [flutter/flutter#130932]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Adding a bottomNavigationBar to an app with `useMaterial3: true` specified in the ThemeData. 

When specifying the bottomNavigationBar property in Scaffold, you can provide either a [BottomNavigationBar](https://api.flutter.dev/flutter/material/BottomNavigationBar-class.html) or a [NavigationBar](https://api.flutter.dev/flutter/material/NavigationBar-class.html). 

If a BottomNavigationBar (Material 2) is provided, no warnings are given that I might want to use the NavigationBar instead because I've enabled Material 3. 

Additionally, API docs for [BottomNavigationBar](https://api.flutter.dev/flutter/material/BottomNavigationBar-class.html) doesn't give any acknowledgement to NavigationBar which is it's new replacement. BottomNavigationBar also comes up first when you Google ["navigation bar flutter"](https://www.google.com/search?q=navigation+bar+flutter&rlz=1C5CHFA_enUS870US870&oq=navigat&aqs=chrome.0.69i59j0i20i263i433i512j69i64j69i57j69i60l3j69i65.1954j0j7&sourceid=chrome&ie=UTF-8) so developers who are looking this up will go for the older Material 2 version instead of the new Material 3 Navigation Bar.

### Proposal

If useMaterial3 is specified in a ThemeData, it can be assumed that developers want to use Material 3 version of widgets. Provide an explicit warning when flutter code specifies an older Material 2 version of widgets that have a Material 3 counterpart available, such as BottomNavigationBar vs NavigationBar. 

Additionally, Material 2 widget documentation should highlight newer Material 3 counterparts, if available.

**Comment by huycozy** [comment]
Thank @bleroux for the reference. I can see the API docs has been updated on master.

>If a BottomNavigationBar (Material 2) is provided, no warnings are given that I might want to use the NavigationBar instead because I've enabled Material 3.

Keeping the issue open for updating the remaining part of proposal.

**Comment by HansMuller** [comment]
Good point about the user's likely intentions. We could detect this kind of thing at run-time (and, as [Bruno points out](https://github.com/flutter/flutter/issues/130932#issuecomment-1643452829), see : https://github.com/flutter/flutter/issues/127216). The style-guide covers producing informational/advisory messages like this: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console

Maybe what's needed is a debugMaterial3Transition flag that enables suggestions about migrating to M3.

**Comment by guidezpl** [comment]
Good idea! I would err on the side of making that `debugMaterial3Transition` true by default for maximum usefulness but also to enable silencing those warnings for those who (incorrectly, might I say) want to use M2 widgets with `useMaterial3: true`.
